doc/source/_templates/searchbox.html

```diff
@@ -20,4 +20,30 @@ <h4>{{ _('Quick search') }}</h4>
     </div>
 </div>
 <script type="text/javascript">$('#searchbox').show(0);</script>
+<div class="well">
+<strong>Docs for {{ release }}<br></strong>
+<a id="other">All versions</a>
+<ul id="versionList" style="display: none;">
+        <script type="text/javascript">
+            var versions = ['dev', '1.15.1', '1.14.5', '1.13.0', '1.12.0', '1.11.0', '1.10.1', '1.9.3', '1.8.1', '1.7.0'];
+            for (i = 0; i < versions.length; i++) {
+
+                if (i == 0) {
+                    base_url = 'https://www.numpy.org/devdocs/index.html'
+                }
+                else {
+                    base_url = 'https://docs.scipy.org/doc/numpy-VERSION'
```

Comment:
More ambitious might be to try to preserve a users' location within the docs when "time travelling" to a different doc version (`np.sqrt` in 1.15 -> `np.sqrt` in 1.13); that said, skimage doesn't do this -- you just end up back at the index of the version you selected regardless of where you were before.

Might be neat, but also would be sensitive to doc restructuring over time causing breakages.

Reply:
Python itself has a [javascript](https://github.com/python/cpython/blob/master/Doc/tools/static/switchers.js) that does this pretty well. It builds a list of possible urls, and uses the first valid one. You can see it in action if you go to [docs.python.org](https://docs.python.org/3/reference/datamodel.html#objects-values-and-types) and click the version button at the top of the page